presto-client/src/main/java/io/prestosql/client/SpnegoHandler.java

```diff
@@ -100,7 +102,16 @@ public SpnegoHandler(
         this.keytab = requireNonNull(keytab, "keytab is null");
         this.credentialCache = requireNonNull(credentialCache, "credentialCache is null");
 
-        kerberosConfig.ifPresent(file -> System.setProperty("java.security.krb5.conf", file.getAbsolutePath()));
+        kerberosConfig.ifPresent(file -> {
+            String newValue = file.getAbsolutePath();
+            String currentValue = System.getProperty("java.security.krb5.conf");
+            checkState(
```

Comment:
This should be `checkArgument` since the problem is with the value of the argument. Or perhaps just use `verify()`.

Reply:
I used `checkState`, because it's not strictly about argument -- the argument to this method is alright, just we cannot apply it because of the state the world (JVM) is in.

`verify` is like assertion, so doesn't seem suitable -- failure may happen due to misconfiguration, using feature that does not exist (2 hive connectors with different kerb settings -- in case of the other place), or (here) because we're embedded in a bigger application that already set something